src/rpc/blockchain.cpp

```diff
@@ -2219,7 +2225,10 @@ static RPCHelpMan savemempool()
         throw JSONRPCError(RPC_MISC_ERROR, "Unable to dump mempool to disk");
     }
 
-    return NullUniValue;
+    UniValue ret(UniValue::VOBJ);
+    ret.pushKV("filename", ((*node_context).args->GetDataDirNet() / "mempool.dat").string());
```

Comment:
We've started using `fs::PathToString` instead of `.string()`

See #22937

Reply:
Done in 09c3395 . Thanks for suggestion.

Reply:
Good catch! This code would do a silent locale conversion on windows after we stop using boost.

Since JSON strings must be UTF-8, would suggest using `.u8string()`. (`PathToString` is meant for internal use and string<->byte roundtrips and is not guaranteed to be UTF-8.)

```c++
ret.pushKV("filename", fs::path((*node_context).args->GetDataDirNet() / "mempool.dat").u8string());
```